[VarDumper] Add maxDepth & maxStringLength display options

nicolas-grekas opened the pull request:
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #16741 |
| License | MIT |
| Doc PR | - |

Takes over #18148 to add display options to html dumps.
Status: needs work

MGDSoft committed:
[VarDumper] Add maxDepth & maxStringLength display options

nicolas-grekas committed:
[VarDumper] Tweak display options implementation

stof commented on src/Symfony/Component/VarDumper/Dumper/HtmlDumper.php:
```diff
@@ -43,7 +43,15 @@ class HtmlDumper extends CliDumper
         'index' => 'color:#1299DA',
+        'str .max-string-length b' => 'color:#A0A0A0',
     );
+    protected $displayOptions = array(
```
should be private

phansys commented on src/Symfony/Component/VarDumper/Dumper/HtmlDumper.php:
```diff
@@ -324,6 +408,15 @@ function isCtrlKey(e) {
+    display:none;
+}
+pre.sf-dump .max-string-length.collapsed .expanded {
+    display:none
```
Missing delimiter.

xabbuh commented on src/Symfony/Component/VarDumper/Dumper/HtmlDumper.php:
```diff
@@ -117,7 +137,8 @@ protected function getDumpHeader()
 
-        $line = <<<'EOHTML'
+        $line = json_encode($this->displayOptions, JSON_FORCE_OBJECT);
+        $line = <<<EOHTML
```
We have to stick with nowdoc here or alternatively have to escape `$` characters below.

fabpot commented:
@nicolas-grekas  Can you finish this one?

fabpot commented:
@nicolas-grekas What's the status of this one?

nicolas-grekas commented:
I'm on it :)

nicolas-grekas commented:
I finally managed to finish this one.

@MGDSoft FYI, I removed the initDepth/maxDepth distinction: it felt strange when playing with the UI. I kept only your initDepth behavior, renamed as maxDepth.

See this fiddle to try the resulting behavior (maxDepth=2):
https://jsfiddle.net/Lz08fgct/

Status: needs review

fabpot commented on src/Symfony/Component/VarDumper/Dumper/HtmlDumper.php:
```diff
@@ -76,6 +82,17 @@ public function setStyles(array $styles)
     /**
+     * Configures display options.
+     *
+     * @param array $displayOptions A map of display options to customize the behavior.
```
no dot at the end

fabpot replied:
ahah, you've been doing that for a long time, working on a PR to fix that :)

fabpot commented:
Thank you @nicolas-grekas.

HeahDude commented:
Nice :)
